lib/fathead/opam/parse.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/python
+
+import os
+from urlparse import urljoin
+
+from bs4 import BeautifulSoup
+
+__target__ = "https://opam.ocaml.org/"
+__module__ = "OPAM"
+
+class Environment(object):
+  """
+  The Environment class is responsible for setting up the directory
+  and retrieving the appropriate files from the web. It is made up of
+  a constructor and a 'setup/0' and 'get_contents/0' member functions.
```

Comment:
And not specify the arity at all? I'm fine with that. It's just become a personal preference really but I suppose we aren't employing any function level polymorphism so I will change it. 😄